cmd/cli/cmd/alpha/manifest-gen/handlers.go

```diff
@@ -0,0 +1,92 @@
+package manifestgen
+
+import (
+	"strings"
+
+	"capact.io/capact/cmd/cli/cmd/alpha/manifest-gen/attributes"
+	"capact.io/capact/cmd/cli/cmd/alpha/manifest-gen/common"
+	"capact.io/capact/cmd/cli/cmd/alpha/manifest-gen/implementations"
+	"capact.io/capact/cmd/cli/cmd/alpha/manifest-gen/interfaces"
+	"capact.io/capact/internal/cli/alpha/manifestgen"
+	"capact.io/capact/pkg/sdk/apis/0.0.1/types"
+	"github.com/pkg/errors"
+	"k8s.io/utils/strings/slices"
+)
+
+type genManifestFn func(opts common.ManifestGenOptions) (manifestgen.ManifestCollection, error)
+
+func generateAttribute(opts common.ManifestGenOptions) (manifestgen.ManifestCollection, error) {
+	files, err := attributes.GenerateAttributeFile(opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "while generating attribute file")
+	}
+	return files, nil
+}
+
+func generateType(opts common.ManifestGenOptions) (manifestgen.ManifestCollection, error) {
+	if slices.Contains(opts.ManifestsType, string(types.InterfaceManifestKind)) {
+		inputTypeManifest, err := interfaces.GenerateInterfaceFile(opts, manifestgen.GenerateInputTypeTemplatingConfig)
+		if err != nil {
+			return nil, errors.Wrap(err, "while generating input type templating config")
+		}
+		if !slices.Contains(opts.ManifestsType, string(types.ImplementationManifestKind)) {
+			outputTypeManifest, err := interfaces.GenerateInterfaceFile(opts, manifestgen.GenerateOutputTypeTemplatingConfig)
+			if err != nil {
+				return nil, errors.Wrap(err, "while generating output type templating config")
+			}
+			inputTypeManifest = mergeManifests(inputTypeManifest, outputTypeManifest)
+		}
+		return inputTypeManifest, nil
+	}
+	files, err := interfaces.GenerateInterfaceFile(opts, manifestgen.GenerateTypeTemplatingConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "while generating type templating config")
+	}
+	return files, nil
+}
+
+func generateInterfaceGroup(opts common.ManifestGenOptions) (manifestgen.ManifestCollection, error) {
+	var files manifestgen.ManifestCollection
+	var err error
+	if slices.Contains(opts.ManifestsType, string(types.InterfaceManifestKind)) {
+		files, err = interfaces.GenerateInterfaceGroupFileFromInterfacePath(opts)
+	} else {
+		files, err = interfaces.GenerateInterfaceGroupFile(opts)
+	}
+
+	if err != nil {
+		return nil, errors.Wrap(err, "while generating interface group templating config")
+	}
+	return files, nil
+}
+
+func generateInterface(opts common.ManifestGenOptions) (manifestgen.ManifestCollection, error) {
+	if slices.Contains(opts.ManifestsType, string(types.TypeManifestKind)) {
+		opts.TypeInputPath = types.ManifestRef{
+			Path:     common.CreateManifestPath(types.TypeManifestKind, opts.ManifestPath) + "-input",
+			Revision: opts.Revision,
+		}
+		if !slices.Contains(opts.ManifestsType, string(types.ImplementationManifestKind)) {
```

Comment:
Remove this if: If Type selected along with Interface + Implementation => also generate Interface output Type